Client/src/ChronologClientImpl.cpp

```diff
@@ -64,6 +64,22 @@ chronolog::ChronologClientImpl::ChronologClientImpl(const std::string &protocol_
                                                                visor_portal_service_provider);
 }
 
+void chronolog::ChronologClientImpl::defineClientIdentity()
+{
+    //uint32_t uid = geteuid(); //TODO: effective uid might be a better choice than login name ...
+
+    char login_name[LOGIN_NAME_MAX];
+    getlogin_r(login_name, LOGIN_NAME_MAX);
```

Comment:
both getlogin() and getlogin_r() fail on WSL. I think it is more reasonable to check uid instead of login username. Users may want to run as another user using sudo which cannot be detected by getlogin_r().

Reply:
I do agree that using effective uid is a better choice ( thus my comment in the code at line 69).

Please, note that we would be using three int32 values to identify the client instance (uid + hostid + pid) . The hash function used in the visor to generate clientId is still the one that takes 2 character strings and generates uint64_t value. I suggest that we revisit ClientId generation mechanism in our next iteration

Reply:
@ibrodkin I am okay with both euid or username. The new changes are fine. But this comment came from the previous code failing in WSL. My idea was to call getpwuid(geteuid()) and get username from pw_name of the returned structure.

I agree with you that the ClientId generation needs to be updated.